presto-hive/src/main/java/com/facebook/presto/hive/security/SqlStandardAccessControl.java

```diff
@@ -788,18 +795,22 @@ private boolean hasAdminOptionForRoles(ConnectorTransactionHandle transaction, C
             return true;
         }
 
-        SemiTransactionalHiveMetastore metastore = getMetastore(transaction);
-        Set<String> rolesWithGrantOption = listApplicableRoles(new PrestoPrincipal(USER, identity.getUser()), (PrestoPrincipal p) -> metastore.listRoleGrants(metastoreContext, p))
-
-                .filter(RoleGrant::isGrantable)
-                .map(RoleGrant::getRoleName)
-                .collect(toSet());
-        return rolesWithGrantOption.containsAll(roles);
+        return getMetastore(transaction)
+                .map(metastore -> {
+                    Set<String> rolesWithGrantOption = listApplicableRoles(new PrestoPrincipal(USER, identity.getUser()), (PrestoPrincipal p) -> metastore.listRoleGrants(metastoreContext, p))
+                            .filter(RoleGrant::isGrantable)
+                            .map(RoleGrant::getRoleName)
+                            .collect(toSet());
+                    return rolesWithGrantOption.containsAll(roles);
+                })
+                .orElse(false);
     }
 
-    private SemiTransactionalHiveMetastore getMetastore(ConnectorTransactionHandle transaction)
+    private Optional<SemiTransactionalHiveMetastore> getMetastore(ConnectorTransactionHandle transaction)
     {
         TransactionalMetadata metadata = hiveTransactionManager.get(transaction);
-        return metadata.getMetastore();
+        // In some scenarios, for example, when a statement in a non-autocommit transaction fails,
+        // the corresponding transaction metadata could be null.
+        return Optional.ofNullable(metadata == null ? null : metadata.getMetastore());
```

Comment:
The idiomatic way to write this would be 
```suggestion
        return Optional.ofNullable(metadata).map(HiveMetadata::getMetastore);
```